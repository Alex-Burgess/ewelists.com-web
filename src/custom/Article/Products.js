import React, { useState, useEffect } from 'react';
import update from 'immutability-helper';
import { API } from "aws-amplify";
// nodejs library to set properties for components
import PropTypes from "prop-types";
// @material-ui/core components
import { makeStyles } from "@material-ui/core/styles";
import Playlist from "@material-ui/icons/PlaylistAdd";
// core components
import GridContainer from "components/Grid/GridContainer.js";
import GridItem from "components/Grid/GridItem.js";
import Card from "components/Card/Card.js";
import CardHeader from "components/Card/CardHeader.js";
import CardBody from "components/Card/CardBody.js";
import Button from "components/CustomButtons/Button.js";
// custom components
import AddPopOut from "./AddPopOut.js";

import styles from "assets/jss/custom/components/article/productsStyle.js";
const useStyles = makeStyles(styles);

export default function Products(props) {
  const { products, lists, isAuthenticated } = props;
  const classes = useStyles();

  const [productDetails, setProductsDetails] = useState({});
  const [addPopouts, setAddPopouts] = useState({});

  useEffect( () => {
    async function getProduct(id) {
      let response;

      try {
        response = await API.get('products', "/" + id);
      } catch (e) {
        console.log("Could not find a product in the products table for id: " + id);
        return false;
      }

      return response;
    }

    async function getProductDetails(products) {
      let productDetails = {};

      for (var ref in products) {
        const productResponse = await getProduct(products[ref]);
        if (productResponse) {
          productDetails[products[ref]] = productResponse;
        }
      }

      setProductsDetails(
        productDetails
      )
    }

    getProductDetails(products);
  }, [products]);

  const handleEditClose = (id) => {
    setAddPopouts({
      addPopouts: update(addPopouts, {
        [id]: {$set: false}
      })
    })
  }

  const handleAddOpen = (id) => {
    setAddPopouts({
      ...addPopouts,
        [id]: true
    })
  }

  const renderAddPopOuts = () => {
    return Object.entries(productDetails).map(
      ([i, product]) =>
        <AddPopOut
          open={addPopouts[product['productId']]
            ? addPopouts[product['productId']]
            : false }
          lists={lists}
          product={product}
          handleClose={handleEditClose}
          key={i}
        />
    )
  }

  return (
    <div className={classes.section}>
      <GridContainer justify="center">
        <GridItem xs={12} sm={10} md={10}>
          <GridContainer>
            {
              Object.entries(productDetails).map(
                ([i, product]) =>
                  <GridItem xs={12} sm={4} md={4} key={i}>
                    <Card plain product className={classes.customProduct}>
                      <CardHeader noShadow image>
                        <a href={product.url} target="_blank" rel="noopener noreferrer">
                          <img src={product.imageUrl} className={classes.productImage} alt=".." />
                        </a>
                        <a href={product.url} target="_blank" rel="noopener noreferrer">
                          <h4 className={classes.cardTitle}>{product.brand}</h4>
                        </a>
                        <div className={classes.description}>
                          {product.details}
                        </div>
                      </CardHeader>
                      <CardBody plain>
                        <a href={product.url} target="_blank" rel="noopener noreferrer">
                          { product.retailer === 'amazon'
                          ? <Button className={classes.amazonButton}>
                              <i className="fab fa-amazon" /> Buy Now
                            </Button>
                          : <Button color="primary" className={classes.button}>
                              Buy Now
                            </Button>
                          }
                        </a>
                        { isAuthenticated
                        ? <Button color="info" className={classes.button} onClick={() => handleAddOpen(product.productId)}>
                             <Playlist /> Add To List
                           </Button>
                        : <a href="/signup" target="_blank" rel="noopener noreferrer">
                            <Button color="info" className={classes.button}>
                             Sign up to Create List
                           </Button>
                         </a>
                        }
                      </CardBody>
                    </Card>
                  </GridItem>
              )
            }
          </GridContainer>
        </GridItem>
      </GridContainer>
      {renderAddPopOuts()}
    </div>
  );
}

Products.propTypes = {
  products: PropTypes.array,
  lists: PropTypes.object,
  isAuthenticated: PropTypes.bool
};
