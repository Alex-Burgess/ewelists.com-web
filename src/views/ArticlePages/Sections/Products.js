import React, { useState, useEffect } from 'react';
import update from 'immutability-helper';
// nodejs library to set properties for components
import PropTypes from "prop-types";
// @material-ui/core components
import { makeStyles } from "@material-ui/core/styles";
import Playlist from "@material-ui/icons/PlaylistAdd";
// core components
import GridContainer from "components/Grid/GridContainer.js";
import GridItem from "components/Grid/GridItem.js";
import Card from "components/Card/Card.js";
import CardHeader from "components/Card/CardHeader.js";
import CardBody from "components/Card/CardBody.js";
import Button from "components/Buttons/Button.js";
// custom components
import AddPopOut from "./AddPopOut.js";
// libs
import { useAppContext } from "libs/contextLib";
import { onError } from "libs/errorLib";
import { getBlogProducts } from "libs/apiLib";

import styles from "assets/jss/material-kit-pro-react/views/articlePages/productsStyle.js";
const useStyles = makeStyles(styles);

export default function Products(props) {
  const classes = useStyles();
  const { products, lists } = props;
  const { isAuthenticated } = useAppContext();
  const [addPopouts, setAddPopouts] = useState({});

  const [data, setProductData] = useState({});
  const [isLoading, setIsLoading] = useState(true);

  useEffect( () => {
    async function getProductData(products){
      try {
        let productDetails = await getBlogProducts(products);
        setProductData(productDetails);
        setIsLoading(false);
      } catch (e) {
        onError(e)
      }
    };

    getProductData(products);
  }, [products]);

  const handleEditClose = (id) => {
    setAddPopouts({
      addPopouts: update(addPopouts, {
        [id]: {$set: false}
      })
    })
  }

  const handleAddOpen = (id) => {
    setAddPopouts({
      ...addPopouts,
        [id]: true
    })
  }

  const renderAddPopOuts = () => {
    return products.map((id, key) =>
      <AddPopOut
        open={addPopouts[id]
          ? addPopouts[id]
          : false }
        lists={lists}
        product={data[id]}
        handleClose={handleEditClose}
        key={key}
      />
    )
  }

  return (
    !isLoading && (
      <div className={classes.section}>
        <GridContainer justify="center">
          <GridItem xs={12} sm={10} md={10}>
            <GridContainer>
              {
                products.map((id, key) =>
                  <GridItem xs={12} sm={4} md={4} key={key}>
                    <Card plain product className={classes.customProduct} data-cy={"product-card-" + data[id].productId}>
                      <CardHeader noShadow image className={classes.customImage}>
                        <a href={data[id].productUrl} target="_blank" rel="noopener noreferrer">
                          <img src={data[id].imageUrl} className={classes.productImage} alt=".." />
                        </a>
                        <a href={data[id].productUrl} target="_blank" rel="noopener noreferrer">
                          <h4 className={classes.cardTitle}>{data[id].brand}</h4>
                        </a>
                        <div className={classes.description}>
                          {data[id].details}
                        </div>
                        <div className={classes.price}>
                          {data[id].price
                            ? <div>
                                Â£ {data[id].price}
                              </div>
                            : null
                          }
                        </div>
                      </CardHeader>
                      <CardBody plain className={classes.customBody}>
                        { isAuthenticated
                        ? <Button color="primary2" className={classes.button} onClick={() => handleAddOpen(data[id].productId)} data-cy="button-add-to-list">
                             <Playlist /> Add To List
                           </Button>
                        : <a href="/signup" target="_blank" rel="noopener noreferrer">
                            <Button color="primary2" className={classes.button}>
                             Sign up to Create List
                           </Button>
                         </a>
                        }
                        <a href={data[id].productUrl} target="_blank" rel="noopener noreferrer">
                          <Button color="primary" className={classes.button}>
                            Buy Now
                          </Button>
                        </a>
                      </CardBody>
                    </Card>
                  </GridItem>
              )
            }
            </GridContainer>
          </GridItem>
        </GridContainer>
        {renderAddPopOuts()}
      </div>
    )
  );
}

Products.propTypes = {
  products: PropTypes.array,
  lists: PropTypes.object
};
