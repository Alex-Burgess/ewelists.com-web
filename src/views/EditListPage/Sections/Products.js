import React, { useState } from 'react';
import update from 'immutability-helper';
import FadeLoader from "react-spinners/FadeLoader";
// nodejs library to set properties for components
import PropTypes from "prop-types";
// @material-ui/core components
import { makeStyles } from "@material-ui/core/styles";
import Tooltip from "@material-ui/core/Tooltip";

// @material-ui icons
import Edit from "@material-ui/icons/Edit";
import Playlist from "@material-ui/icons/PlaylistAdd";
// core components
import Table from "components/Table/Table.js";
import Button from "components/Buttons/Button.js";
import Card from "components/Card/Card.js";
import CardBody from "components/Card/CardBody.js";
// Sections
import SectionEdit from "./EditProductPopOut.js";

import styles from "assets/jss/material-kit-pro-react/views/editListPage/productsStyle.js";
const useStyles = makeStyles(styles);

export default function SectionProducts(props) {
  const classes = useStyles();
  const { mobile, listId, products, loading } = props;

  const [editPopouts, setEditPopouts] = useState({});

  const handleEditClose = (id) => {
    setEditPopouts({
      editPopouts: update(editPopouts, {
        [id]: {$set: false}
      })
    })
  }

  const handleEditClickOpen = (id) => {
    setEditPopouts({
      ...editPopouts,
        [id]: true
    })
  }

  const renderDesktopProductView = () => {
    return (
      <Table
        tableHead={[
          "",
          "GIFTS",
          "QTY",
          "RESERVED",
          ""
        ]}
        tableData={
          renderProducts()
        }
        tableShopping
        customHeadCellClasses={[
          classes.textCenter,
          classes.textCenter,
          classes.textCenter,

        ]}
        customHeadClassesForCells={[2, 3, 4]}
        customCellClasses={[
          classes.tdImage,
          classes.tdName,
          classes.tdNumber + " " + classes.textCenter,
          classes.tdNumber + " " + classes.textCenter,
          classes.textCenter
        ]}
        customClassesForCells={[0, 1, 2, 3, 4]}
      />
    )
  }

  const renderMobileProductView = () => {
    return (
      <Table
        tableHead={[
          "MANAGE GIFTS"
        ]}
        tableData={
          renderMobileProducts()
        }
        tableShopping
        customHeadCellClasses={[
          classes.textCenter
        ]}
        customHeadClassesForCells={[0]}
      />
    )
  }

  const renderProducts = () => {
    const allproducts = Object.entries(products).map(
        ([key, p]) =>
              renderProduct(p['productId'], p['productUrl'], p['imageUrl'], p['brand'], p['details'], p['quantity'], p['reserved'] + p['purchased'], p['price'])
      );

    return allproducts
  }

  const renderMobileProducts = () => {
    const allproducts = Object.entries(products).map(
      ([key, p]) =>
            renderMobileProduct(p['productId'], p['productUrl'], p['imageUrl'], p['brand'], p['details'], p['quantity'], p['reserved'] + p['purchased'])
    )

    return allproducts
  }

  const renderProduct = (productId, productUrl, imageUrl, brand, details, quantity, reserved, price) => {
    return (
      [
      <div className={classes.imgContainer} key={1}>
        <button onClick={() => handleEditClickOpen(productId)} className={classes.undoButton} data-cy="link-image">
          <img src={imageUrl} alt="..." className={classes.img} />
        </button>
      </div>,
      <span key={1}>
        <button onClick={() => handleEditClickOpen(productId)} className={classes.brand} data-cy="link-brand">
          {brand}
        </button>
        <br />
        <small className={classes.tdNameSmall}>
          {details}
        </small>
        {price
          ? <div className={classes.price}>
              £ {price}
            </div>
          : null
        }
      </span>,
      <span key={1}>
        {quantity}
      </span>,
      <span key={1}>
        {reserved}
      </span>,
      <span>
        <Tooltip
          id="edit"
          title="Edit item"
          placement="left"
          classes={{ tooltip: classes.tooltip }}
        >
          <Button link className={classes.actionButton} onClick={() => handleEditClickOpen(productId)} data-cy="link-edit">
            <Edit />
          </Button>
        </Tooltip>
      </span>
      ]
    )
  }

  const renderMobileProduct = (productId, productUrl, imageUrl, brand, details, quantity, reserved, price) => {
    return (
      [
          <div className={classes.textCenter}>
            <div className={classes.imgContainer} key={1}>
              <button onClick={() => handleEditClickOpen(productId)} className={classes.undoButton} data-cy="link-image">
                <img src={imageUrl} alt="..." className={classes.img} />
              </button>
            </div>
            <button onClick={() => handleEditClickOpen(productId)} className={classes.undoButton} data-cy="link-brand">
              <h4 className={classes.brand}>
                {brand}
              </h4>
            </button>
            <br />
            <small className={classes.mobileDescription}>
              {details}
            </small>
            {price
              ? <div className={classes.price}>
                  £ {price}
                </div>
              : null
            }
            <br />
            <small className={classes.quantities}>
              Quantity: {quantity}
            </small>
            <br />
            <small className={classes.quantities}>
              Reserved: {reserved}
            </small>
            <br />
            <Tooltip
              id="edit"
              title="Edit item"
              placement="left"
              classes={{ tooltip: classes.tooltip }}
            >
              <Button link className={classes.actionButton} onClick={() => handleEditClickOpen(productId)} data-cy="link-edit">
                <Edit />
              </Button>
            </Tooltip>
          </div>
      ]
    )
  }

  const renderEditPopOuts = () => {
    return Object.entries(products).map(
      ([key, product]) =>
          <SectionEdit
            open={editPopouts[product['productId']]
              ? editPopouts[product['productId']]
              : false }
            listId={listId}
            product={product}
            handleClose={handleEditClose}
            deleteProductFromState={props.deleteProductFromState}
            updateProductToState={props.updateProductToState}
            key={key}
          />
    )
  }

  return (
    <div className={classes.section}>
      <div className={classes.container}>
        <Card plain>
          <CardBody plain>
            {
              mobile
              ? renderMobileProductView()
              : renderDesktopProductView()
            }
            {loading
              ? <div className={classes.loading}>
                  <FadeLoader
                    size={50}
                    color={"#9a9a9a"}
                    loading={true}
                  />
                </div>
              : null
            }
            <div className={classes.addItemButton}>
              <Button round color="primary" onClick={() => props.switchToAddProduct(1)} >
                <Playlist /> Add Item
              </Button>
            </div>

          </CardBody>
        </Card>
        {renderEditPopOuts()}
      </div>
    </div>
  );
}

SectionProducts.propTypes = {
  loading: PropTypes.bool,
  mobile: PropTypes.bool,
  listId: PropTypes.string,
  products: PropTypes.object
};
