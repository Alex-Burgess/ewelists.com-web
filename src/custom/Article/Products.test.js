import * as React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import Products from './Products';

const products = [
  {brand: 'John Lewis and Partners', url: 'https://www.johnlewis.com/john-lewis-partners-geneva-large-weekend-holdall/blue/p1807673', price: '£75.00', description: 'John Lewis & Partners Geneva Large Weekend Holdall, Blue.', img: 'https://johnlewis.scene7.com/is/image/JohnLewis/234172546?$rsp-pdp-port-1440$'},
  {brand: "Burt's Bees", url: 'https://www.amazon.co.uk/Burts-Bees-Percentage-Overnight-Ultra-Conditioning/dp/B07C3BLZRN', price: '£6.99', description: "Burt's Bees 100 Percentage Natural Overnight Intensive Lip Treatment, Ultra-Conditioning Lip Care, 7.08 g.", img: 'https://images-na.ssl-images-amazon.com/images/I/71yUvdw8UIL._SX679_.jpg'},
  {brand: 'Camelbak', url: 'https://www.amazon.co.uk/Camelbak-53622-CamelBak-eddy-75L/dp/B00NTYIHNQ', price: '£11.95', description: 'BPA Free Eddy Outdoor Bottle.', img: 'https://images-na.ssl-images-amazon.com/images/I/61J1m1AOrVL._SY879_.jpg'},
];

test.skip('Renders renderer component.', () => {

  const tree = renderer
    .create(
      <Router>
        <Products
          products={products}
        />
      </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
