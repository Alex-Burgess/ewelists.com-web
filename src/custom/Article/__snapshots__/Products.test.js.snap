// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders renderer component. 1`] = `
<div
  className="makeStyles-section-1"
>
  <div
    className="MuiGrid-root makeStyles-grid-9  MuiGrid-container MuiGrid-justify-xs-center"
  >
    <div
      className="MuiGrid-root makeStyles-grid-113  MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-10 MuiGrid-grid-md-10"
    >
      <div
        className="MuiGrid-root makeStyles-grid-9  MuiGrid-container"
      >
        <div
          className="MuiGrid-root makeStyles-grid-113  MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 MuiGrid-grid-md-4"
        >
          <div
            className="makeStyles-card-114 makeStyles-cardPlain-115 makeStyles-cardProduct-122 makeStyles-customProduct-8"
          >
            <div
              className="makeStyles-cardHeader-129 makeStyles-cardHeaderImage-131 makeStyles-noShadow-132"
            >
              <a
                href="https://www.johnlewis.com/john-lewis-partners-geneva-large-weekend-holdall/blue/p1807673"
                rel="noopener noreferrer"
                target="_blank"
              >
                <img
                  alt=".."
                  className="makeStyles-productImage-3"
                  src="https://johnlewis.scene7.com/is/image/JohnLewis/234172546?$rsp-pdp-port-1440$"
                />
              </a>
              <a
                href="https://www.johnlewis.com/john-lewis-partners-geneva-large-weekend-holdall/blue/p1807673"
                rel="noopener noreferrer"
                target="_blank"
              >
                <h4
                  className="makeStyles-cardTitle-4"
                >
                  John Lewis and Partners
                </h4>
              </a>
              <div
                className="makeStyles-description-5"
              >
                John Lewis & Partners Geneva Large Weekend Holdall, Blue.
              </div>
              <div
                className="makeStyles-priceContainer-6"
              >
                <span
                  className="makeStyles-price-7"
                >
                   
                  £75.00
                </span>
              </div>
            </div>
            <div
              className="makeStyles-cardBody-141 makeStyles-cardBodyPlain-143"
            >
              <button
                className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-148 makeStyles-primary-150"
                disabled={false}
                onBlur={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                <span
                  className="MuiButton-label"
                >
                  <i
                    className="fab fa-amazon"
                  />
                   Buy Now
                </span>
              </button>
              <button
                className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-148 makeStyles-info-152"
                disabled={false}
                onBlur={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                <span
                  className="MuiButton-label"
                >
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M14 10H2v2h12v-2zm0-4H2v2h12V6zm4 8v-4h-2v4h-4v2h4v4h2v-4h4v-2h-4zM2 16h8v-2H2v2z"
                    />
                  </svg>
                   Add To List
                </span>
              </button>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root makeStyles-grid-113  MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 MuiGrid-grid-md-4"
        >
          <div
            className="makeStyles-card-114 makeStyles-cardPlain-115 makeStyles-cardProduct-122 makeStyles-customProduct-8"
          >
            <div
              className="makeStyles-cardHeader-129 makeStyles-cardHeaderImage-131 makeStyles-noShadow-132"
            >
              <a
                href="https://www.amazon.co.uk/Burts-Bees-Percentage-Overnight-Ultra-Conditioning/dp/B07C3BLZRN"
                rel="noopener noreferrer"
                target="_blank"
              >
                <img
                  alt=".."
                  className="makeStyles-productImage-3"
                  src="https://images-na.ssl-images-amazon.com/images/I/71yUvdw8UIL._SX679_.jpg"
                />
              </a>
              <a
                href="https://www.amazon.co.uk/Burts-Bees-Percentage-Overnight-Ultra-Conditioning/dp/B07C3BLZRN"
                rel="noopener noreferrer"
                target="_blank"
              >
                <h4
                  className="makeStyles-cardTitle-4"
                >
                  Burt's Bees
                </h4>
              </a>
              <div
                className="makeStyles-description-5"
              >
                Burt's Bees 100 Percentage Natural Overnight Intensive Lip Treatment, Ultra-Conditioning Lip Care, 7.08 g.
              </div>
              <div
                className="makeStyles-priceContainer-6"
              >
                <span
                  className="makeStyles-price-7"
                >
                   
                  £6.99
                </span>
              </div>
            </div>
            <div
              className="makeStyles-cardBody-141 makeStyles-cardBodyPlain-143"
            >
              <button
                className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-148 makeStyles-primary-150"
                disabled={false}
                onBlur={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                <span
                  className="MuiButton-label"
                >
                  <i
                    className="fab fa-amazon"
                  />
                   Buy Now
                </span>
              </button>
              <button
                className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-148 makeStyles-info-152"
                disabled={false}
                onBlur={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                <span
                  className="MuiButton-label"
                >
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M14 10H2v2h12v-2zm0-4H2v2h12V6zm4 8v-4h-2v4h-4v2h4v4h2v-4h4v-2h-4zM2 16h8v-2H2v2z"
                    />
                  </svg>
                   Add To List
                </span>
              </button>
            </div>
          </div>
        </div>
        <div
          className="MuiGrid-root makeStyles-grid-113  MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-4 MuiGrid-grid-md-4"
        >
          <div
            className="makeStyles-card-114 makeStyles-cardPlain-115 makeStyles-cardProduct-122 makeStyles-customProduct-8"
          >
            <div
              className="makeStyles-cardHeader-129 makeStyles-cardHeaderImage-131 makeStyles-noShadow-132"
            >
              <a
                href="https://www.amazon.co.uk/Camelbak-53622-CamelBak-eddy-75L/dp/B00NTYIHNQ"
                rel="noopener noreferrer"
                target="_blank"
              >
                <img
                  alt=".."
                  className="makeStyles-productImage-3"
                  src="https://images-na.ssl-images-amazon.com/images/I/61J1m1AOrVL._SY879_.jpg"
                />
              </a>
              <a
                href="https://www.amazon.co.uk/Camelbak-53622-CamelBak-eddy-75L/dp/B00NTYIHNQ"
                rel="noopener noreferrer"
                target="_blank"
              >
                <h4
                  className="makeStyles-cardTitle-4"
                >
                  Camelbak
                </h4>
              </a>
              <div
                className="makeStyles-description-5"
              >
                BPA Free Eddy Outdoor Bottle.
              </div>
              <div
                className="makeStyles-priceContainer-6"
              >
                <span
                  className="makeStyles-price-7"
                >
                   
                  £11.95
                </span>
              </div>
            </div>
            <div
              className="makeStyles-cardBody-141 makeStyles-cardBodyPlain-143"
            >
              <button
                className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-148 makeStyles-primary-150"
                disabled={false}
                onBlur={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                <span
                  className="MuiButton-label"
                >
                  <i
                    className="fab fa-amazon"
                  />
                   Buy Now
                </span>
              </button>
              <button
                className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-button-148 makeStyles-info-152"
                disabled={false}
                onBlur={[Function]}
                onDragLeave={[Function]}
                onFocus={[Function]}
                onKeyDown={[Function]}
                onKeyUp={[Function]}
                onMouseDown={[Function]}
                onMouseLeave={[Function]}
                onMouseUp={[Function]}
                onTouchEnd={[Function]}
                onTouchMove={[Function]}
                onTouchStart={[Function]}
                tabIndex={0}
                type="button"
              >
                <span
                  className="MuiButton-label"
                >
                  <svg
                    aria-hidden="true"
                    className="MuiSvgIcon-root"
                    focusable="false"
                    role="presentation"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M14 10H2v2h12v-2zm0-4H2v2h12V6zm4 8v-4h-2v4h-4v2h4v4h2v-4h4v-2h-4zM2 16h8v-2H2v2z"
                    />
                  </svg>
                   Add To List
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
